apis/youtube/api.go

```diff
@@ -0,0 +1,94 @@
+package youtube
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	part   = "snippet"
+	fields = "items.id,items.snippet.title,items.snippet.description,items.snippet.publishedAt"
+)
+
+// API allows to query data from the Twitter APIs
+type API struct {
+	endpoint string
+	apiKey   string
+	client   *http.Client
+}
+
+// NewAPI builds a new API instance
+func NewAPI(apiKey string) *API {
+	return &API{
+		endpoint: "https://www.googleapis.com/youtube/v3",
+		apiKey:   apiKey,
+		client:   &http.Client{},
+	}
+}
+
+// runRequest runs the given request, and returns the response body.
+// If the request contains error, or the server answers with a status code different than 200,
+// the method will return an error
+func (api *API) runRequest(req *http.Request) ([]byte, error) {
+	// Add the headers
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-goog-api-key", fmt.Sprintf(api.apiKey))
```

Comment:
I think the formatting here can be removed
```suggestion
	req.Header.Add("X-goog-api-key", api.apiKey)
```